### PR TITLE
update dependencies and tests to latest

### PR DESCRIPTION
**Pull request by hassanbazzi**
Just part of the changes needed. WIP.

### CODE REVIEW

**Comment by hassanbazzi** [comment]
@developit Do you have any ideas what's going on in that properties file? I'm assuming auto-generated?

**Comment by developit** [comment]
It's the name mappings for uglify. I believe they changed the format in 3.x - we'll have to modify it since right now it's generating invalid names.

File: package.json [hunk 1]
```diff
@@ -107,18 +107,18 @@
...
     "phantomjs-prebuilt": "^2.1.7",
     "rimraf": "^2.5.3",
-    "rollup": "^0.48.2",
+    "rollup": "^0.56.3",
     "rollup-plugin-babel": "^3.0.2",
-    "rollup-plugin-memory": "^2.0.0",
+    "rollup-plugin-memory": "^3.0.0",
     "rollup-plugin-node-resolve": "^3.0.0",
-    "sinon": "^2.2.0",
+    "sinon": "^4.4.2",
```
Comment (marvinhagemeister) [inline package.json]: That's awesome! Thanks a lot for your PR 🎉 

The CI reports quite a few deprecation messages like:

```
INFO: 'sinon.reset is deprecated and will be removed from the public API in a future version of sinon.'
```

Perhaps we can fix these too?

File: config/properties.json [hunk 2]
```diff
@@ -1,27 +1,53 @@
...
-    }
-  },
-  "vars": {
-    "cname": -1,
-    "props": {}
-  }
-}
\ No newline at end of file
+	"props": {
+		"props": {
```
Comment (marvinhagemeister) [inline config/properties.json]: This seems to be a mistake :)

Comment (hassanbazzi) [reply config/properties.json]: Yup. I need to look at this again.

File: package.json (hunk 1 above)
Comment (hassanbazzi) [reply package.json]: will do

**Comment by marvinhagemeister** [comment]
@hassanbazzi Just saw that you were active here again 🎉 Do you feel like you have enough time to address the review comments? I'm happy to merge your PR instead of #1032 once the comments are addressed 👍

**Comment by mkxml** [comment]
Closing this since #1032 was merged

**Pull request closed without merging**